frame/support/src/storage/generator/nmap.rs

```diff
@@ -0,0 +1,524 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+	hash::{StorageHasher, Twox128},
+	storage::{
+		self,
+		types::{HasKeyPrefix, HasReversibleKeyPrefix, KeyGenerator, ReversibleKeyGenerator},
+		unhashed, PrefixIterator, StorageAppend, TupleToEncodedIter,
+	},
+	Never,
+};
+use codec::{Decode, Encode, EncodeLike, FullCodec};
+#[cfg(not(feature = "std"))]
+use sp_std::prelude::*;
+
+/// Generator for `StorageNMap` used by `decl_storage`.
```

Comment:
Is this really only used by `decl_` macros? 🤔

Reply:
I think he copy/paste from other generator module like generator/double_map.rs

the trait is currently use by decl_storage to implement, and it is hidden from doc. But now it is also used by frame/support/src/storage/types/* 
```suggestion
/// Generator for `StorageNMap` used by `decl_storage` and storage types.
```